server/base/src/main/java/org/apache/accumulo/server/util/FileUtil.java

```diff
@@ -522,15 +506,15 @@ public static Map<TabletFile,FileInfo> tryToGetFirstAndLastRows(ServerContext co
   }
 
   public static WritableComparable<Key> findLastKey(ServerContext context,
-      Collection<TabletFile> mapFiles) throws IOException {
+      TableConfiguration tableConf, Collection<TabletFile> mapFiles) throws IOException {
 
     Key lastKey = null;
 
     for (TabletFile file : mapFiles) {
       FileSystem ns = context.getVolumeManager().getFileSystemByPath(file.getPath());
       FileSKVIterator reader = FileOperations.getInstance().newReaderBuilder()
-          .forFile(file.getPathStr(), ns, ns.getConf(), context.getCryptoService())
-          .withTableConfiguration(context.getConfiguration()).seekToBeginning().build();
+          .forFile(file.getPathStr(), ns, ns.getConf(), tableConf.getCryptoService())
+          .withTableConfiguration(tableConf).seekToBeginning().build();
```

Comment:
I wonder if `FileOperations` should change, something like:

```
FileOperations.getInstance().newReaderBuilder().table(TableConfiguration)
     .file(String, FileSystem, Configuration);
```

Reply:
FileOperations is a mess. There is definitely room for improvement.